packages/mongodb-access-adapter/datagrip-access-adapter/src/test/kotlin/com/mongodb/jbplugin/accessadapter/datagrip/IntegrationTest.kt

```diff
@@ -0,0 +1,144 @@
+/**
+ * Test extension that allows us to test with the IntelliJ environment
+ * without spinning up the whole IDE. Also, sets up a MongoDB instance
+ * that can be queried.
+ */
+
+package com.mongodb.jbplugin.accessadapter.datagrip
+
+import com.intellij.database.dataSource.DatabaseDriverManagerImpl
+import com.intellij.database.dataSource.LocalDataSource
+import com.intellij.database.dataSource.validation.DatabaseDriverValidator.createDownloaderTask
+import com.intellij.database.psi.DataSourceManager
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.EDT
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.progress.EmptyProgressIndicator
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.ProjectManager
+import com.intellij.testFramework.junit5.RunInEdt
+import com.intellij.testFramework.junit5.TestApplication
+import com.intellij.util.ui.EDT
+import com.mongodb.jbplugin.accessadapter.MongoDbDriver
+import com.mongodb.jbplugin.accessadapter.datagrip.adapter.DataGripMongoDbDriver
+import org.junit.jupiter.api.extension.*
+import org.testcontainers.containers.MongoDBContainer
+import org.testcontainers.junit.jupiter.Testcontainers
+import org.testcontainers.lifecycle.Startables
+
+import java.nio.file.Files
+import java.util.*
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+
+/**
+ * Represents what version of MongoDB we support in the plugin.
+ */
+enum class MongoDbVersion(val versionString: String) {
+    LATEST("7.0.9"),
+    ;
+}
+
+/**
+ * Annotation to be used in the test, at the class level.
+ *
+ * @see com.mongodb.jbplugin.accessadapter.datagrip.adapter.DataGripMongoDbDriverTest
+ */
+@TestApplication
+@RunInEdt(allMethods = true, writeIntent = true)
+@ExtendWith(IntegrationTestExtension::class)
+@Testcontainers(parallel = false)
+annotation class IntegrationTest(val mongodb: MongoDbVersion = MongoDbVersion.LATEST, val sharded: Boolean = false)
+
+/**
+ * Extension implementation. Must not be used directly.
+ */
+internal class IntegrationTestExtension : BeforeAllCallback,
+    AfterAllCallback,
+    ParameterResolver {
+    private val namespace = ExtensionContext.Namespace.create(IntegrationTestExtension::class.java)
+    private val containerKey = "CONTAINER"
+    private val projectKey = "PROJECT"
+    private val driverKey = "DRIVER"
+    private val versionKey = "VERSION"
+
+    override fun beforeAll(context: ExtensionContext?) {
+        val annotation = context!!.requiredTestClass.getAnnotation(IntegrationTest::class.java)
+        val container = MongoDBContainer("mongo:${annotation.mongodb.versionString}-jammy")
```

Comment:
Just a clarifying question: container usage = We're not planning to run tests on platforms other than Linux?

Reply:
I'm not sure I follow, they should run on any environment with Docker/Podman (I'm running them on my Mac with Podman right now). Do you see any concern here?